### ISSUE TITLE
[feature] Self-destruct command

### ISSUE THREAD

**Issue by vivithecanine** [superseriousbusiness/gotosocial#1697]
### Is your feature request related to a problem ?

Mastodon offers a feature that is helpful when switching between different activitypub technologies called `tootctl self-destruct`. This command sends out account delete broadcasts for all accounts known to the server, making all other servers defederate from this host. It would be helpful for gts users to have this to prevent "burning" a domain by using gts who may need to use something like Mastodon for now. GTS does not currently for example support mastodon -> GTS account migration.

### Describe the solution you'd like.

Provide a `gotosocial destroy` command that sends an ActivityPub  Account Delete for every existing account known to the local server to all known instances.

### Describe alternatives you've considered.

If you aren't using a base domain for your instance, you may be able to switch between subdomains to clean up after gts, but you will still have burned a subdomain in the process.

**Comment by tsmethurst** [comment]
Good idea! We gotta investigate whether this will actually allow someone to deploy another fedi instance on the same domain afterwards. My spidey sense tells me that it wouldn't work because most of the time when you send a Delete to a remote, it just stubs the account but doesn't actually remove it from the db (this is true of mastodon, iirc, and is also what GtS does). So it may not allow to avoid 'burning' domains, but it's still a good idea to provide a 'correct' neighborly way of shutting down an instance that doesn't leave as much clutter in remote instance's DBs.

**Comment by hl0dwig** [comment]
I'm wondering what happens, for a domain, if we don't send any such signal, and we just delete the federation service?

I know Mastodon has a internal feature to stop sending messages to a domain when it's not reachable for a certain amount of time. But I'm thinking it's something only true for Mastodon as it's not listed in the ActivityPub protocol (I didn't find anything related to see in the [official doc](https://www.w3.org/TR/activitypub)) 

For example I'm running right now a GtS server on domain A.COM
I stop the service, then all other servers will still try to reach the A.COM

- What happens for other GtS servers? They will try to do so without ever stopping?
- For Mastodon servers, they'll stop doing so, soon or later

I'm thinking about this, because I'm gonna create a new instance of GtS behind a new domain name, and will abandon the actual one... but I'm kind of concerned about the futur of this abandoned domain name, and servers that will still try to federate with it desperately! (yeah I care about virtual stuff 💔)

**Comment by eallion** [comment]
I have migrated my instance to a new domain, but I can still find information about the old domain instance in the federated universe, although it is outdated.

**Comment by hl0dwig** [comment]
that was also my fear @eallion 

I guess I will first delete each of my post manually from my actual domain before moving to the new one and abandoning the actual one...

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/1697.
